tests/e2e/tests/build/build-app-shell.ts

```diff
@@ -0,0 +1,129 @@
+import { ng, npm } from '../../utils/process';
+import { expectFileToMatch, writeFile } from '../../utils/fs';
+import { expectToFail } from '../../utils/utils';
+import { updateJsonFile } from '../../utils/project';
+import { stripIndent } from 'common-tags';
+
+
+export default function () {
+  return Promise.resolve()
+    .then(() => updateJsonFile('.angular-cli.json', configJson => {
+      const app = configJson['apps'][0];
+      app['appShell'] = {
+        app: '1',
+        route: 'shell'
+      };
+      configJson['apps'].push({
+        platform: 'server',
+        root: 'src',
+        outDir: 'dist-server',
+        assets: [
+          'assets',
+          'favicon.ico'
+        ],
+        index: 'index.html',
+        main: 'main.server.ts',
+        test: 'test.ts',
+        tsconfig: 'tsconfig.server.json',
+        testTsconfig: 'tsconfig.spec.json',
+        prefix: 'app',
+        styles: [
+          'styles.scss'
+        ],
+        scripts: [],
+        environmentSource: 'environments/environment.ts',
+        environments: {
+          dev: 'environments/environment.ts',
+          prod: 'environments/environment.prod.ts'
+        }
+      });
+    }))
+    .then(() => writeFile('src/app/app.module.ts', stripIndent`
+      import { BrowserModule } from '@angular/platform-browser';
+      import { NgModule } from '@angular/core';
+      import { RouterModule } from '@angular/router';
+
+      import { AppComponent } from './app.component';
+
+      @NgModule({
+        imports: [
+          BrowserModule.withServerTransition({ appId: 'appshell-play' }),
+          RouterModule
+        ],
+        declarations: [AppComponent],
+        bootstrap: [AppComponent]
+      })
+      export class AppModule { }
+    `))
+    .then(() => writeFile('src/app/app.component.html', stripIndent`
+      Hello World
+      <router-outlet></router-outlet>
+    `))
+    .then(() => writeFile('src/tsconfig.server.json', stripIndent`
+      {
+        "extends": "../tsconfig.json",
+        "compilerOptions": {
+          "outDir": "../out-tsc/app",
+          "baseUrl": "./",
+          "module": "commonjs",
+          "types": []
+        },
+        "exclude": [
+          "test.ts",
+          "**/*.spec.ts"
+        ],
+        "angularCompilerOptions": {
+          "entryModule": "app/app.server.module#AppServerModule"
+        }
+      }
+    `))
+    .then(() => writeFile('src/main.server.ts', stripIndent`
+      export {AppServerModule} from './app/app.server.module';
+    `))
+    .then(() => writeFile('src/app/app.server.module.ts', stripIndent`
```

Comment:
Having the `AppServerModule` in the same place that normal `AppModule` is breaking basic generate commands. For example, `ng g component hello` fails with this error : `Error: More than one module matches.`

Reply:
I agree in general practice, but not in this case for a build test scenario where there will be no generating.